pkg/httpd/chat.go

```diff
@@ -123,6 +139,14 @@ func (h *chat) processChatMessages(
 			h.sendSessionMessage(id, answer, messageID, "message", openAIConn.IsReasoning)
 		case answer := <-openAIConn.DoneCh:
 			h.sendSessionMessage(id, answer, messageID, "finish", false)
+			runes := []rune(answer)
+			if len(runes) > 100 {
+				answer = string(runes[:100])
+			}
+			conversation.Context = append(conversation.Context, QARecord{
+				Question: conversation.Question,
+				Answer:   answer,
+			})
 			return answer
 		}
 	}
```

Comment:
None found. Please ensure the above code is updated to reflect recent changes in coding conventions or best practices.